docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,172 @@
+  For example, after the redesign, hive_cluster_deployments_running_transition_seconds will be reported if and only if metricsConfig.clusterDeploymentRelatedMetrics.metrics[hive_cluster_deployments_running_transition_seconds].minimumDuration is not empty, however hive_cluster_deployment_install_job_duration_seconds, which is a non-optional metric that would support minimumDuration, will be reported even if metricsConfig.clusterDeploymentRelatedMetrics.metrics[hive_cluster_deployment_install_job_duration_seconds].minimumDuration is not specified.
+- Additionally, we would be removing the fixed labels that can be propagated from the cluster deployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new clusterDeploymentRelatedMetrics, sufficient documentation should help though.
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveconfig, you can either provide the new configurations (aka `clusterDeploymentRelatedMetrics`) or the old way of hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+Note that only the old design will support the mapping of camelCase key to metric for minimumDuration, the new way will require full metric names only.
+
+#### Failure modes
+
+These situations will cause a panic().
+- clusterDeploymentRelatedMetrics.metrics[].name doesn't exist as a metric
+- clusterDeploymentRelatedMetrics.metrics[].name is a metric without duration but minimumDuration was specified.
+- a metric is defined across multiple groupings and the customizations defined contradict. For ex, following hiveConfig will cause panic because hive will be confused whether to parse `minimumDuration` for `hive_foo_seconds` to be 10m or 1h.
+  ```
+  HiveConfig:
+    spec:
+      metricsConfig:
+        clusterDeploymentRelatedMetrics:
+        - metrics:
+          - hive_foo_seconds
+          minimumDuration: 10m
+        - metrics:
+          - hive_foo_seconds
+          - hive_foo_gauge
+          minimumDuration: 1h
+  ```
+- additionalClusterDeploymentLabels conflict with existing fixed label for that metric
+- if deprecated methods are used along with the new clusterDeploymentRelatedMetrics. You can either choose the _old_ way or the _new_ way.
+
+
+### Risks and Mitigations
+The biggest risk is the sheer number of metrics that are going to be affected. Roughly 29 metrics that we report right now can be classified as related to clusterDeployment.
```

Comment:
Not to be pedantic, but why is this a risk? Because of the amount of code we're going to touch? The disruption to the user when they cut over? It shouldn't pose a risk from a design/architecture or execution standpoint.

Reply:
I think it is a risk to touch too many metrics, you have to ensure no metric changes its behaviour unexpectedly - this would require thorough testing from both the development and QE standpoint. I've added some clarification, though I don't mind dropping this from the section if you insist.